core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1158,9 +1158,14 @@ class KafkaController(val config : KafkaConfig, zkUtils: ZkUtils, val brokerStat
     @throws(classOf[Exception])
     def handleNewSession() {
       info("ZK expired; shut down all controller components and try to re-elect")
-      onControllerResignation()
-      inLock(controllerContext.controllerLock) {
-        controllerElector.elect
+      if (controllerElector.getControllerID() != config.brokerId) {
+        onControllerResignation()
+        inLock(controllerContext.controllerLock) {
+          controllerElector.elect
+        }
+      } else {
```

Comment:
hi junrao, thanks for the review. This bug is found in the 0.9, it seems KAFKA-1387 not working well for this case,  the method also throwing the ZKNodeExistException。 And I also found if the ephemeral node is created by previous session, when creating the /brokers/ids/[id] it will also throw  ZKNodeExistException in our product env， like the issue:  https://issues.apache.org/jira/browse/KAFKA-4277.